core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/unfold.kt

```diff
@@ -0,0 +1,23 @@
+package org.jetbrains.kotlinx.dataframe.impl.api
+
+import org.jetbrains.kotlinx.dataframe.AnyCol
+import org.jetbrains.kotlinx.dataframe.DataColumn
+import org.jetbrains.kotlinx.dataframe.api.CreateDataFrameDsl
+import org.jetbrains.kotlinx.dataframe.api.asColumnGroup
+import org.jetbrains.kotlinx.dataframe.api.asDataColumn
+import org.jetbrains.kotlinx.dataframe.api.isPrimitive
+import org.jetbrains.kotlinx.dataframe.columns.ColumnKind
+import org.jetbrains.kotlinx.dataframe.typeClass
+
+@PublishedApi
+internal fun <T> DataColumn<T>.unfoldImpl(skipPrimitive: Boolean, body: CreateDataFrameDsl<T>.() -> Unit): AnyCol {
+    return when (kind()) {
+        ColumnKind.Group, ColumnKind.Frame -> this
+        else -> when {
+            skipPrimitive && isPrimitive() -> this
```

Comment:
You can't. Have a look `unfold primitive`. skipPrimitive = false is needed to make it work, and skipPrimitive = true is needed to avoid unpacking for example a column of String to `ColumnGroup, size: Int`, the same as we do for toDataFrame with overloads

Reply:
Yes, but can you take a look at the `isPrimitive()` function? That function also returns true when you run in on a collection and an array. My suggestion was only to rename the `isPrimitive()` function.